include/seastar/http/common.hh

```diff
@@ -30,6 +30,9 @@ namespace seastar {
 namespace http {
 namespace internal {
 output_stream<char> make_http_chunked_output_stream(output_stream<char>& out);
+// The len parameter defines the maximum number of bytes to be written. After the
+// stream is closed, the len is updated with the actual number of bytes written.
```

Comment:
I don't understand how learning the length **after** the write completed is "content_length". It's confusing when this code sits in an HTTP implementation, because "content length" in HTTP is specifically about knowing the length before the write starts :-(

Reply:
Well, the use-case is supposed to be like this:

- caller sets content_length and body writing callback
- client code sends http headers including the content-length one
- client code calls caller's callback to write body into the stream
- caller's callback is buggy and closes the stream before writing promised amount of bytes

What should the client code do in that case? This patch's answer is -- after the body is written check that the contract is fulfilled and abort with exception otherwise. In order to "check that the contract is fulfilled" the body writing sink reports back the amount of bytes it wrote into the outgoing stream